.travis.yml

```diff
@@ -26,28 +26,32 @@ script:
 - docker tag todo-app:prod legendarypan/todo-app:prod
 - docker push legendarypan/todo-app:prod
```

Comment:
Not new here, but we should move this building/pushing your prod image into a deploy only step, otherwise you'll be pushing an image on every PR; in particular that could mean two parallel builds causing the wrong branch to be deployed with Azure.

We can move this stage to a before_deploy lifecycle step:
```yml
...
- echo "All Tests Have Been Completed"
before_deploy:
- docker login --username "$docker_login" --password "$docker_pass"
- docker build --target production --tag todo-app:prod .
- docker tag todo-app:prod legendarypan/todo-app:prod
- docker push legendarypan/todo-app:prod

deploy:
...
```

Reply:
got it. thanks!